tools/bqtag/bqtag/bqtag.py

```diff
@@ -0,0 +1,800 @@
+        if default_column_tag.strip() != "":
+            default_column_tag = default_column_tag.strip().lower()
+            if default_column_tag in self.policy_tags:
+                for column in json_schema:
+                    self._process_default_tag(
+                                                column=column,
+                                                tag=self.policy_tags[
+                                                    default_column_tag
+                                                ]
+                                            )
+
+        return json_schema
+
+
+    # Internal function to add the default tag to schema.
+    def _process_default_tag(self, column: dict, tag: str) -> None:
+        """
+        Recursively tag all columns that do not have a tag
+
+        :param column: column data
+        :param tag: Default tag
+        """
+
+        if column["type"].upper() == "RECORD":
+            for sub_column in column["fields"]:
+                self._process_default_tag(column=sub_column, tag=tag)
+        else:
+            if "policyTags" not in column:
+                column["policyTags"] = {"names": []}
+                column["policyTags"]["names"].append(tag)
+
+
+    # Read schema file and cereate policy-tag column map
+    def _create_tag_column_map(self, schema: str) -> dict:
+
+        """
+        Read table schema and map columns to tags.
+
+        :param schema: Schema of the table
+        :return Map of columns to tag
+        """
+
+        schema_json = json.loads(schema)
+
+        tag_to_columns = dict()
+        tag_to_columns_mapped = dict()
+
+        # update tag_to_columns
+        self._rec_create_tag_column_map(
+                                            items=schema_json,
+                                            tag_to_columns=tag_to_columns
+                                        )
+
+        #map tag_to_columns key to tag_map and populate tag
+        for k,v in tag_to_columns.items():
+
+            if k in self.policy_tags_rev:
+                tag_to_columns_mapped[
+                    self.policy_tags_rev[k]
+                ] = copy.deepcopy(v)
+            else:
+                print("Policy Tag", k, "not found in map file.")
+
+        return copy.deepcopy(tag_to_columns_mapped)
+
+
+    def _rec_create_tag_column_map(
+                                    self,
+                                    items: list,
+                                    tag_to_columns: dict,
+                                    parent="",
+                                    paren_mode=""
+                                ) -> None:
+
+        """
+        Recursive Function to map columns to tags.
+
+        :param items: List of columns
+        :param tag_to_columns: map of tag to columns
+        :param parent: Parent column in case of nested column
+        :param parent_mode: Mode of the parent column
+        :return None
+        """
+
```

Comment:
Remove extra blank line

Reply:
Changed to:

```
    def _rec_create_tag_column_map(
                                    self,
                                    items: list,
                                    tag_to_columns: dict,
                                    parent="",
                                    paren_mode=""
                                ) -> None:
        """
        Recursive Function to map columns to tags.

        :param items: List of columns
        :param tag_to_columns: map of tag to columns
        :param parent: Parent column in case of nested column
        :param parent_mode: Mode of the parent column
        :return: None
        """
        for item in items:
```